backend/src/service/scan-mode.service.ts

```diff
@@ -0,0 +1,73 @@
+import JoiValidator from '../web-server/controllers/validators/joi.validator';
+import ScanModeRepository from '../repository/scan-mode.repository';
+import OIAnalyticsMessageService from './oia/oianalytics-message.service';
+import { ScanModeCommandDTO, ValidatedCronExpression } from '../../../shared/model/scan-mode.model';
+import { scanModeSchema } from '../web-server/controllers/validators/oibus-validation-schema';
+import SouthCacheRepository from '../repository/south-cache.repository';
+import { ScanMode, ScanModeCommand } from '../model/scan-mode.model';
+import { validateCronExpression } from './utils';
+import OIBusEngine from '../engine/oibus-engine';
+
+export default class ScanModeService {
+  constructor(
+    protected readonly validator: JoiValidator,
+    private scanModeRepository: ScanModeRepository,
+    private southCacheRepository: SouthCacheRepository,
+    private oIAnalyticsMessageService: OIAnalyticsMessageService,
+    private oibusEngine: OIBusEngine
+  ) {}
+
+  findAll(): Array<ScanMode> {
+    return this.scanModeRepository.findAll();
+  }
+
+  findById(id: string): ScanMode | null {
+    const scanMode = this.scanModeRepository.findById(id);
+    if (scanMode) {
+      return scanMode;
+    } else {
+      return null;
+    }
```

Comment:
this is weird this should already be the return type of the repo method, you can just proxy to the repo call